cupy/_core/core.pyx

```diff
@@ -238,11 +238,22 @@ cdef class ndarray:
         if not runtime._is_hip_environment:  # CUDA
             if stream is None:
                 stream = runtime.streamLegacy
-            elif not isinstance(stream, int) or stream < -1 or stream == 0:
+            elif not isinstance(stream, int) or stream < -1:
+                # DLPack does not accept 0 as a valid stream, but there is a
+                # bug in PyTorch that exports the default stream as 0, which
+                # renders the protocol unusable, we will accept a 0 value
+                # meanwhile.
                 raise ValueError(
                     f'On CUDA, the valid stream for the DLPack protocol is -1,'
                     f' 1, 2, or any larger value, but {stream} was provided')
             if curr_stream_ptr == 0:
+                if not _util.DLPACK_ACCEPT_STREAM_ZERO:
+                    raise ValueError(
+                        'stream 0 is passed by library XXX, which is not '
+                        'accepted per the DLPack protocol. Please report this '
+                        'problem to the upstream library, and in the meantime '
+                        'work around this by setting the env var '
+                        'CUPY_DLPACK_ACCEPT_STREAM_ZERO=1')
```

Comment:
Sorry, this line (249) is wrong: We should check the `stream` argument, not CuP's current stream (`curr_stream_ptr`).
```suggestion
            if stream == 0:
                if not _util.DLPACK_ACCEPT_STREAM_ZERO:
                    raise ValueError(
                        'stream 0 is passed by library XXX, which is not '
                        'accepted per the DLPack protocol. Please report this '
                        'problem to the upstream library, and in the meantime '
                        'work around this by setting the env var '
                        'CUPY_DLPACK_ACCEPT_STREAM_ZERO=1')
            if curr_stream_ptr == 0:
```